__tests__/app.test.js

```diff
@@ -333,3 +333,22 @@ describe('DELETE /api/comments/:comment_id', () => {
         })
     })
 })
+
+describe('GET /api/users', () => {
+    test ('200: returns an array of user objects, each of which having a username, name and avatar_url properties', () => {
+      return request(app).get('/api/users').expect(200).then(({ body }) => {
+          const { users } = body
+          expect(users).toHaveLength(4);
+          users.forEach((user) => {
+              expect(user).toMatchObject({
+                  username: expect.any(String),
+                  name: expect.any(String),
+                  avatar_url: expect.any(String)
+              })
+          })
+      })
+  })
+  test('404: returns an error for an invalid endpoint', () => {
```

Comment:
You don't need to test this for every possible iteration of a bad endpoint. If you have one test for a bad url that is enough.